services/server/src/server/services/compiler/lambda-with-fallback/SolcLambdaWithLocalFallback.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  CompilerOutput,
+  ISolidityCompiler,
+  JsonInput,
+} from "@ethereum-sourcify/lib-sourcify";
+import logger from "../../../../common/logger";
+import { SolcLambda } from "../lambda/SolcLambda";
+import { SolcLocal } from "../local/SolcLocal";
+
+export class SolcLambdaWithLocalFallback implements ISolidityCompiler {
+  private solcLambda = new SolcLambda();
+  private solcLocal = new SolcLocal();
+
+  public async compile(
+    version: string,
+    solcJsonInput: JsonInput,
+    forceEmscripten: boolean = false
+  ): Promise<CompilerOutput> {
+    let compilerOutput: CompilerOutput;
+    try {
+      compilerOutput = await this.solcLambda.compile(
+        version,
+        solcJsonInput,
+        forceEmscripten
+      );
+    } catch (e) {
+      logger.error(
```

Comment:
I think that here we need to switch on a specific error not all the errors

Reply:
(I pushed the new commit before I saw your comment update)

The problem with this is that I cannot really find out which error code is given by AWS in the case the response is too big. I was checking the docs and they were not very helpful (https://docs.aws.amazon.com/lambda/latest/api/API_InvokeWithResponseStreamCompleteEvent.html). 

We could also try to find a contract whose compiler output is bigger than the response stream limit of 20 MB. I'm just not sure how to go about that.

Reply:
I'm thinking... Maybe on the lambda side you can find the output size and if it's more than 20mb send a specific response. Could this work?